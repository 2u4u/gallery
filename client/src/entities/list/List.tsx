import { FileType } from 'shared/model/reducers/appReducer';
import './index.scss';

type Props = {
    file: FileType;
    handleOpenModal: () => void;
}

function List({ file, handleOpenModal }: Props) {
    let content = null;

    switch (file.type) {
        case 'image':
            content = (
                <div className='list' onClick={handleOpenModal}>
                    <div className='list__img' style={{ backgroundImage: 'url(' + file.url + ')'}} />
                    <span>{file.name}</span>
                </div>);
            break;
        case 'audio':
        case 'video':
            content = (
                <div className='list' onClick={handleOpenModal}>
                    <div className='list__icon'>
                        <img src={`./img/${file.type}.svg`} alt={file.name} />
                        <div className='list__duration'>{file.duration}</div>
                    </div>
                    <span>{file.name}</span>
                </div>
            )
            break;
        default:
            content = (
                <div className='list' onClick={handleOpenModal}>
                    <div className='list__icon'>
                        <img src={`./img/unknown.svg`} alt='not defined' />
                    </div>
                    <span>{file.name}</span>
                </div>
            )
            break;
    }
    
    return (content);
}

export default List;
