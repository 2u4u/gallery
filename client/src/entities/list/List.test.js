import renderer from 'react-test-renderer';
import List from './List';

describe('List component ', () => {
    it('matches snapshot', () => {
        const file = {
            _id: 'someid',
            type: 'image',
            name: 'test image',
            url: 'https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png',
            duration: 'undefined',
            date: Date,
        }
        const component = renderer.create(
            <List file={file} handleOpenModal={() => false}/>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
