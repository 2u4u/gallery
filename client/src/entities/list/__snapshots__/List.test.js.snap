// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component  matches snapshot 1`] = `
<div
  className="list"
  onClick={[Function]}
>
  <div
    className="list__img"
    style={
      Object {
        "backgroundImage": "url(https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png)",
      }
    }
  />
  <span>
    test image
  </span>
</div>
`;
